packages/eslint-plugin/tests/rules/no-lifecycle-call.test.ts

```diff
@@ -0,0 +1,671 @@
+
+    `@Directive({
+      selector: 'test'
+    })
+    class Test {
+      test() {
+        super.ngDoCheck();
+      }
+    }`,
+
+    `@Directive({
+      selector: 'test'
+    })
+    class Test {
+      test() {
+        super.ngOnChanges();
+      }
+    }`,
+
+    `@Directive({
+      selector: 'test'
+    })
+    class Test {
+      test() {
+        super.ngOnInit();
+      }
+    }`,
+    // #endregion Directives
+
+    // #region Injectables
+    `@Injectable()
+    class Test {
+      test() {
+        super.ngAfterContentChecked();
+      }
+    }`,
+
+    `@Injectable()
+    class Test {
+      test() {
+        super.ngAfterContentInit();
+      }
+    }`,
+
+    `@Injectable()
+    class Test {
+      test() {
+        super.ngAfterViewChecked();
+      }
+    }`,
+
+    `@Injectable()
+    class Test {
+      test() {
+        super.ngAfterViewInit();
+      }
+    }`,
+
+    `@Injectable()
+    class Test {
+      test() {
+        super.ngDoCheck();
+      }
+    }`,
+
+    `@Injectable()
+    class Test {
+      test() {
+        super.ngOnChanges();
+      }
+    }`,
+
+    `@Injectable()
+    class Test {
+      test() {
+        super.ngOnInit();
+      }
+    }`,
+    // #endregion Injectables
+
+    // #region Pipes
+    `@Pipe({
+      name: 'test'
+    })
+    class Test {
+      test() {
+        super.ngAfterContentChecked();
+      }
+    }`,
+
+    `@Pipe({
+      name: 'test'
+    })
+    class Test {
+      test() {
+        super.ngAfterContentInit();
+      }
+    }`,
+
+    `@Pipe({
+      name: 'test'
+    })
+    class Test {
+      test() {
+        super.ngAfterViewChecked();
+      }
+    }`,
+
+    `@Pipe({
+      name: 'test'
+    })
+    class Test {
+      test() {
+        super.ngAfterViewInit();
+      }
+    }`,
+
+    `@Pipe({
+      name: 'test'
+    })
+    class Test {
+      test() {
+        super.ngDoCheck();
+      }
+    }`,
+
+    `@Pipe({
+      name: 'test'
+    })
+    class Test {
+      test() {
+        super.ngOnChanges();
+      }
+    }`,
+
+    `@Pipe({
+      name: 'test'
+    })
+    class Test {
+      test() {
+        super.ngOnInit();
+      }
+    }`,
+    // #endregion Pipes
+
+    // should succeed if explicitly calling multiple non lifecycle methods
+    `@Component({
+      selector: 'test'
+    })
+    class Test {
+      test(): void {
+        this.ng4fterViewChecked();
+        this.ngOnChange$();
+      }
+    }
+    @Directive({
+      selector: 'test'
+    })
+    class TestDirective {
+      test(): void {
+        this.ngOnD3stroy();
+        this.ngOn1nit();
+      }
+    }`,
+  ],
+  invalid: [
+    // #region Components
+    convertAnnotatedSourceToFailureCase({
+      annotatedSource: `@Component({
+        selector: 'test'
+      })
+      class Test {
+        test() {
+          this.ngAfterContentChecked();
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        }
+      }`,
+      description: 'it should fail if explicitly calling ngAfterContentChecked',
```

Comment:
Similarly could we keep the `description`s as the first things here so that it is more readable overall?

Reply:
Oops, sry. I've committed the requested changes.

<h6>I have a habit of follow `object-literal-sort-keys` rule :stuck_out_tongue_closed_eyes:</h6>